.github/actions/auto-merge-pull-request/action.yml

```diff
@@ -0,0 +1,37 @@
+name: Automatically Merge Pull Request
+description: >
+  Merges the current pull request automatically if the "merge-and-resolve-jira-issue-if-checks-succeed" or
+  "merge-if-checks-succeed" label is present.
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Scripts
+      shell: pwsh
+      run: |
+        (Resolve-Path "${{ github.action_path }}/../../../Scripts").Path >> $Env:GITHUB_PATH
+
+    - name: Check Mergeability
+      id: check-mergeability
+      # Fork PR runs won't have permissions to remove labels, nor do we want to allow auto-merging them.
+      if: github.event.pull_request.head.repo.fork == false
+      uses: Lombiq/GitHub-Actions/.github/actions/check-pull-request-labels@issue/OSOE-477
+      with:
+        label1: "merge-if-checks-succeed"
+        label2: "merge-and-resolve-jira-issue-if-checks-succeed"
+
+    - name: Merge Pull Request
+      if: steps.check-mergeability.outputs.contains-label == 'true'
+      shell: pwsh
+      run: |
+        $url = "https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/merge"
```

Comment:
Can we add something like `Create-Github-Repo-Api-Url' with $Repository and $RelativeUrl parameters, and use it here, in `Initialize-IssueParameters.ps1` and in `cancel-workflow` action?

As an alternative, we could create a set of `Invoke-Github-*-Api` commands to keep GH API handling code in one place. Eg.: Invoke-Github-Repost-Pulls-Merge`, `Invoke-Github-Repost-Actions-Cancel`, etc.

Here is some PS GH API implementation:
https://github.com/microsoft/PowerShellForGitHub
https://github.com/pcgeek86/PSGitHub/

Reply:
I figured factoring out the header creation makes the most sense: The bigger part of the URLs vary depending on the data/operation (the "https://api.github.com/repos/$repository" is the common root), and the `Invoke-WebRequest` varies by the method and if it needs to send a body. So, I think we'd just make things differently complicated on the same level and in the end wouldn't really gain much.

However, using an external PS module altogether can make this easier, and since as you point out they exist, I wouldn't go into doing something similar ourselves. I've created an issue for this because I wouldn't tackle it in this unrelated PR: https://github.com/Lombiq/GitHub-Actions/issues/125